src/MasterChef.sol

```diff
@@ -51,46 +49,45 @@ contract MasterChef is Ownable2StepUpgradeable, IMasterChef {
      * @param veMoe The address of the VeMOE contract.
      * @param factory The address of the rewarder factory.
      * @param treasuryShare The share of the rewards that will be sent to the treasury.
-     * @param futureFundingShare The share of the rewards that will be sent to the future funding.
-     * @param teamShare The share of the rewards that will be sent to the team.
-     */
-    constructor(
-        IMoe moe,
-        IVeMoe veMoe,
-        IRewarderFactory factory,
-        uint256 treasuryShare,
-        uint256 futureFundingShare,
-        uint256 teamShare
-    ) {
+     */
+    constructor(IMoe moe, IVeMoe veMoe, IRewarderFactory factory, uint256 treasuryShare) {
         _disableInitializers();
 
-        if (treasuryShare + futureFundingShare + teamShare > Constants.PRECISION) revert MasterChef__InvalidShares();
+        if (treasuryShare > Constants.PRECISION) revert MasterChef__InvalidShares();
 
         _moe = moe;
         _veMoe = veMoe;
         _rewarderFactory = factory;
 
         _treasuryShare = treasuryShare;
-        _futureFundingShare = futureFundingShare;
-        _teamShare = teamShare;
     }
 
     /**
      * @dev Initializes the MasterChef contract.
      * @param initialOwner The initial owner of the contract.
      * @param treasury The initial treasury.
-     * @param futureFunding The initial future funding vesting contract.
-     * @param team The initial team vesting contract.
-     */
-    function initialize(address initialOwner, address treasury, address futureFunding, address team)
-        external
-        initializer
-    {
+     * @param futureFunding The address of the future funding vesting contract.
+     * @param team The address of the team vesting contract.
+     * @param futureFundingAmount The amount of MOE tokens to pre-mint for the future funding vesting contract.
+     * @param teamAmount The amount of MOE tokens to pre-mint for the team vesting contract.
+     */
+    function initialize(
+        address initialOwner,
+        address treasury,
+        address futureFunding,
+        address team,
+        uint256 futureFundingAmount,
+        uint256 teamAmount
+    ) external reinitializer(2) {
         __Ownable_init(initialOwner);
 
         _setTreasury(treasury);
-        _setFutureFunding(futureFunding);
-        _setTeam(team);
+
+        uint256 mintedToFutureFunding =
+            futureFundingAmount > 0 ? _moe.mint(address(futureFunding), futureFundingAmount) : 0;
+        uint256 mintedToTeam = teamAmount > 0 ? _moe.mint(address(team), teamAmount) : 0;
+
+        require(mintedToFutureFunding == futureFundingAmount && mintedToTeam == teamAmount, "MasterChef: mint failed");
```

Comment:
we use revert in this contract everywhere else, maybe worth changing for consistency?